source/tryGoal/tryGoal_service.js

```diff
@@ -15,6 +15,50 @@ services.factory('ConfigAPI', ['$resource', function($resource) {
     });
 }]);
 
+services.factory('agentHttpInterceptor', ['agentMemory', '$rootScope', function(agentMemory, $rootScope) {
+    
+    $rootScope.apiOutput = new Array();
+
+    return {
+
+        'request': function(config) {
+            
+            if (agentMemory.tryGoal && !config.url.endsWith('.html') && 
+            (config.url.endsWith('query') || config.url.endsWith('response'))) {
+                var log = {type: 'request'};
+                log.method = config.method;
+                log.url = config.url;      
+                log.params = config.params;  
+                log.data = config.data;   
+                log.headers = config.headers;    
+                $rootScope.apiOutput.push(log);
+            }
+            return config;
+        },
+    
+        'response': function(response) {
+            var log;
+            if (response.data && response.data.apiLog) {
+                log = response.data.apiLog;
+                $rootScope.apiOutput.push(log.request);
+                $rootScope.apiOutput.push(log.response);
+
+            } else if (agentMemory.tryGoal && !response.config.url.endsWith('.html') && 
```

Comment:
!config.url.endsWith('.html') is that needed or is it covered by the line below?

Reply:
Yes agreed as above.